prusti-specs/src/export_spec_rewriter.rs

```diff
@@ -0,0 +1,334 @@
+}
+
+// generate macro for export based on given identifier and item
+fn generate_macro_for_export<T: ToTokens>(
+    macro_ident: syn::Ident,
+    macro_content: T,
+) -> TokenStream {
+    quote_spanned! {macro_ident.span() =>
+        #[macro_export]
+        #[prusti::export_spec_macro]
+        macro_rules! #macro_ident {
+            () => {
+                #macro_content
+            };
+        }
+    }
+}
+
+// helper macro for getting the value of attribute with given name in an item
+// attribute must be in this format: #[name(value)]
+// if attribute is not present then path is not changed
+// @param: $path_mut_ref must be a mutable reference of Path, i.e. &mut syn::Path / mut syn::Path
+macro_rules! get_prusti_path_in_attr {
+    ($item: ident, $attr_str: literal, $path_mut_ref: ident) => {
+        if let Some(export_path_attr) = $item.attrs.iter().find(|attr| {
+            attr.path.segments.len() == 2
+                && &attr.path.segments.last().unwrap().ident.to_string() == $attr_str
+        }) {
+            for tt in export_path_attr.tokens.to_owned() {
+                if let TokenTree::Group(group) = tt {
+                    $path_mut_ref = syn::parse2(group.stream()).unwrap();
+                }
+            }
+        }
+    };
+}
+
+// expand the module path recorded when traversing AST
+fn expand_mod_path_in_macro(
+    path: &syn::Path,
+    mut tokens: TokenStream,
+    uses_same_level: &TokenStream,
+) -> TokenStream {
+    match &path.segments.empty_or_trailing() {
+        true => tokens,
+        false => {
+            for seg in path.segments.to_owned().into_iter().rev() {
+                tokens = quote!(
+                    mod #seg {
+                        #uses_same_level
+                        #tokens
+                    }
+                );
+            }
+            tokens
+        }
+    }
+}
+
+fn expand_type_path_in_macro(
+    path: &syn::Path,
+    tokens: TokenStream,
+    uses_same_level: &TokenStream,
+    type_path: &syn::Path,
+) -> TokenStream {
+    let type_path_macro_ident = generate_macro_ident(type_path.to_owned());
+    match type_path.segments.empty_or_trailing() {
+        true => expand_mod_path_in_macro(path, tokens, uses_same_level),
+        false => expand_mod_path_in_macro(
+            path,
+            quote!(
+                $crate::#type_path_macro_ident!();
+                #tokens
+            ),
+            uses_same_level,
+        ),
+    }
+}
+
+pub fn rewrite_export_spec(path: &syn::Path, item: &syn::Item) -> syn::Result<TokenStream> {
+    let mut macros_tokens = TokenStream::new();
+    let mod_path = syn::Path {
+        leading_colon: None,
+        segments: syn::punctuated::Punctuated::new(),
+    };
+    match item {
+        syn::Item::Fn(item_fn) => {
+            rewrite_fn(
+                path,
+                mod_path,
+                item_fn,
+                &mut macros_tokens,
+                &TokenStream::new(),
+            );
```

Comment:
Make these functions return a `TokenStream` instead of taking a mutable reference to the one you create here? The `match` block can be wrapped in `Ok(...)` and it can be the last expression in the method so the `return` is not necessary. All the `{ ... }` braces in this `match` can be omitted.